### PR TITLE
FIX: Resolve numeric overflow in drift estimation node

### PR DESCRIPTION
**Pull request by oesteban**
Addresses the issue reported by @psadil with his proposed solution 1 (considering the slope and intercept).

cc/ @effigies for a correctness check.

Resolves: #1315.

### CODE REVIEW

**Comment by oesteban** [comment]
Let's roll it out and come back to this if @effigies disagrees.

File: mriqc/interfaces/diffusion.py [hunk 1]
```diff
@@ -662,10 +662,17 @@ def _run_interface(self, runtime):
...
+                - intercept
+            )
             full_img.__class__(
-                (full_img.get_fdata() * fitted[np.newaxis, np.newaxis, np.newaxis, :]).astype(
-                    full_img.header.get_data_dtype()
-                ),
+                corrected.astype(full_img.header.get_data_dtype()),
                 full_img.affine,
                 full_img.header,
             ).to_filename(self._results['out_full_file'])
```
Comment (effigies) [inline mriqc/interfaces/diffusion.py:678]: Sorry for the delay. No, you just want:

```python
full_img.__class__(
    full_img.get_fdata() * fitted[np.newaxis, np.newaxis, np.newaxis, :],
    full_img.affine
    full_img.header
).to_filename(self._results['out_full_file'])
```

Comment (oesteban) [reply mriqc/interfaces/diffusion.py:678]: I assume that applies the slope and intercept when generating the spatialimage object, correct?

Comment (effigies) [reply mriqc/interfaces/diffusion.py:678]: Yes, it will calculate a new slope and intercept based on the min/max values of the data array.

**Pull request merged**